### PR TITLE
Add script to evaluate face detection by WIDERFace

### PR DESCRIPTION
**Pull request by WanliZhong**
The script is modified based on [WiderFace-Evaluation](https://github.com/wondervictor/WiderFace-Evaluation).

- [x] Add dataset for evaluation: [WIDERFace](http://shuoyang1213.me/WIDERFACE/)
- [x] Report yunet accuracy

### CODE REVIEW

File: tools/eval/README.md [hunk 1]
```diff
@@ -53,3 +54,48 @@ Run evaluation with the following command:
 python eval.py -m mobilenet -d imagenet -dr /path/to/imagenet
 ```
 
+## WIDERFace
+
+You need to add the following additional dependencies:
+```shell
+pip install scipy
```
Comment (fengyuentau) [inline tools/eval/README.md:61]: Put this along with `tqdm`

File: tools/eval/README.md [hunk 2]
```diff
@@ -53,3 +54,48 @@ Run evaluation with the following command:
...
+```
+
+### Prepare data
+
+Please visit http://shuoyang1213.me/WIDERFACE to download the WIDER Face dataset [Validation Images](https://huggingface.co/datasets/wider_face/resolve/main/data/WIDER_val.zip), [Testing Images](https://huggingface.co/datasets/wider_face/resolve/main/data/WIDER_test.zip), [Face annotations](http://shuoyang1213.me/WIDERFACE/support/bbx_annotation/wider_face_split.zip) and [eval_tools](http://shuoyang1213.me/WIDERFACE/support/eval_script/eval_tools.zip). Organize files as follow:
```
Comment (fengyuentau) [inline tools/eval/README.md:66]: I think we do not need `eval_tools`

File: tools/eval/README.md [hunk 3]
```diff
@@ -53,3 +54,48 @@ Run evaluation with the following command:
...
+
+
+```shell
+$ tree -L 2 /path/to/widerface
+.
+├── eval_tools
```
Comment (fengyuentau) [inline tools/eval/README.md:72]: no need to have `eval_tools`

File: tools/eval/datasets/widerface.py [hunk 4]
```diff
@@ -0,0 +1,309 @@
...
+
+    def __getitem__(self, index):
+        img = cv.imread(self.img_list[index])
+        event, name = self.img_list[index].split(os.sep)[-2:]
+        return event, name, img
+
+    def eval(self, model):
+        results_list = dict()
+        # forward
+        for event_name, img_name, img in tqdm.tqdm(self):
```
Comment (fengyuentau) [inline tools/eval/datasets/widerface.py:287]: Can you provide a screenshot on the output through the whole process? I suggest to print the model name as well.

**Review by fengyuentau — APPROVED** [review]
Looks good to me 👍 Please see below for some minor changes. By the way, can you update `models/face_detection_yunet/README.md` with the accuracy?

File: tools/eval/README.md (hunk 1 above)
Comment (WanliZhong) [reply tools/eval/README.md:61]: OK, I will update it.

File: tools/eval/datasets/widerface.py (hunk 4 above)
Comment (WanliZhong) [reply tools/eval/datasets/widerface.py:287]: I forgot to output the model name, I will add it in the next update.

File: models/face_detection_yunet/README.md [hunk 5]
```diff
@@ -7,6 +7,15 @@ Notes:
 - For details on training this model, please visit https://github.com/ShiqiYu/libfacedetection.train.
 - This ONNX model has fixed input shape, but OpenCV DNN infers on the exact shape of input image. See https://github.com/opencv/opencv_zoo/issues/44 for more information.
 
+Results of accuracy evaluation with [tools/eval](../../tools/eval).
+
+| Models      | Easy AP | Medium AP | Hard AP | 
+|-------------|---------|-----------|---------|
+| YuNet       | 0.8498  | 0.8384    | 0.7357  |
+| YuNet quant | 0.7751  | 0.8145    | 0.7312  |
```
Comment (kaingwade) [inline models/face_detection_yunet/README.md:15]: Medium AP is much higher than Easy AP for YuNet quant?

Comment (fengyuentau) [reply models/face_detection_yunet/README.md:15]: @WanliZhong You can doublecheck with the offcial eval tools of WiderFace

Comment (WanliZhong) [reply models/face_detection_yunet/README.md:15]: The result of the original script is also the same result.
image

**Pull request merged**